tools/dependencies/CMakeLists.txt

```diff
@@ -0,0 +1,130 @@
+cmake_minimum_required(VERSION 3.20.2)
+cmake_policy(SET CMP0135 NEW)
+project(OpenSnDependencies)
+include(ExternalProject)
+include(ProcessorCount)
+ProcessorCount(JOBS)
+if(NOT JOBS OR JOBS EQUAL 0)
+  set(JOBS 4)
+endif()
+
+if(NOT DEFINED CMAKE_INSTALL_PREFIX OR CMAKE_INSTALL_PREFIX STREQUAL "")
+  message(FATAL_ERROR "You must set CMAKE_INSTALL_PREFIX explicitly. For example:\n"
+                      "  cmake -DCMAKE_INSTALL_PREFIX=/path/to/install ..")
+endif()
+message(STATUS "CMAKE_INSTALL_PREFIX is set to: ${CMAKE_INSTALL_PREFIX}")
+
+set(OPENSN_ENV_DIR "${CMAKE_INSTALL_PREFIX}/bin")
+set(ENV_SCRIPT "${OPENSN_ENV_DIR}/set_opensn_env.sh")
+
+find_package(MPI REQUIRED)
+
+option(ENABLE_BOOST   "Install Boost"   ON)
+option(ENABLE_LUA     "Install Lua"     ON)
```

Comment:
Should lua be on?

Reply:
For now, I think it should be. I also need to add Python. I'll revise to check for all of the required packages and install them if not found.